### PR TITLE
fix: use current logrus config when logrus is created internally (#6234)

### PR DESCRIPTION
**Pull request by yu-croco**
Signed-off-by: yu-croco <yuki.kita22@gmail.com>

This PR is for https://github.com/argoproj/argo-cd/issues/6234 .
Some of the log lines are printed as text even if I configured as `--logFormat=json`. This is because some of `logrus.New()` are not configured format and level.

[logrus](https://github.com/sirupsen/logrus) does not have method to return current log format, so I save the logger configuration on env when cli package methods( `SetLogLevel`, `SetLogFormat` ) are called. And I use the env when I create logrus.

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [x] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [x] Does this PR require documentation updates?
* [x] I've updated documentation as required by this PR.
* [x] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

File: util/cli/cli.go [hunk 1]
```diff
@@ -145,21 +146,21 @@ func ReadAndConfirmPassword() (string, error) {
...
+
+	log.SetFormatter(utillog.CreateFormatter(logFormat))
 }
 
 // SetLogLevel parses and sets a logrus log level
 func SetLogLevel(logLevel string) {
 	level, err := log.ParseLevel(logLevel)
 	errors.CheckError(err)
-	log.SetLevel(level)
+	os.Setenv(common.EnvLogLevel, level.String())
```
Comment (yu-croco) [inline util/cli/cli.go:163]: only methods in cli package can set `EnvLogFormat` and `EnvLogLevel`.

File: util/log/logrus.go [hunk 2]
```diff
@@ -2,14 +2,56 @@ package log
...
+	return formatType
+}
+
+func createLogLevel() logrus.Level {
+	level, err := logrus.ParseLevel(os.Getenv(common.EnvLogLevel))
+	if err != nil {
+		level = logrus.InfoLevel
+	}
+	return level
+}
```
Comment (yu-croco) [inline util/log/logrus.go:57]: Validation for `EnvLogLevel` is done when it is set on `SetLogLevel` method, `util/cli/cli.go`.
So I think I do not need to raise error, like `errors.CheckError(err)` (maybe error does not happen here) ... 🤔

**Comment by yu-croco** [comment]
I will fix failed CI 🙏

File: util/log/logrus.go (hunk 2 above)
Comment (alexmt) [reply util/log/logrus.go:59]: agree

**Review by alexmt — APPROVED** [review]
Thank you. LGTM

**Pull request merged**